source/sema.h

```diff
@@ -1511,11 +1511,15 @@ class sema
     )
         -> bool
     {
-        std::vector<error_entry> devnull;
-        auto& errors = emit_errors ? this->errors : devnull;
+        const std::function emit_error = [this](source_position pos, std::string_view msg) {
+            errors.emplace_back(pos, msg);
+        };
+        const std::function skip_error = [](source_position, std::string_view){};
+
+        const auto handle_error = emit_errors ? emit_error : skip_error;
```

Comment:
I chose to use two lambdas to make the `emit_errors == false` more explicitly no-op.